Oh wow, AppVeyor failed with a weird error!?

```
FAIL: test_env_var_debug (test.test_asyncio.test_base_events.BaseEventLoopTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\cpython\lib\test\test_asyncio\test_base_events.py", line 771, in test_env_var_debug
    sts, stdout, stderr = assert_python_ok('-E', '-c', code)
  File "C:\projects\cpython\lib\test\support\script_helper.py", line 157, in assert_python_ok
    return _assert_python(True, *args, **env_vars)
  File "C:\projects\cpython\lib\test\support\script_helper.py", line 143, in _assert_python
    res.fail(cmd_line)
  File "C:\projects\cpython\lib\test\support\script_helper.py", line 84, in fail
    err))
AssertionError: Process return code is 3221225477
command line: ['C:\\projects\\cpython\\PCbuild\\win32\\python.exe', '-X', 'faulthandler', '-I', '-E', '-c', 'import asyncio\nloop = asyncio.get_event_loop()\nprint(loop.get_debug())']
stdout:
---
---
stderr:
---
Windows fatal exception: access violation
Current thread 0x00000184 (most recent call first):
  File "C:\projects\cpython\lib\asyncio\events.py", line 41 in __init__
  File "C:\projects\cpython\lib\asyncio\selector_events.py", line 260 in _add_reader
  File "C:\projects\cpython\lib\asyncio\selector_events.py", line 117 in _make_self_pipe
  File "C:\projects\cpython\lib\asyncio\selector_events.py", line 66 in __init__
  File "C:\projects\cpython\lib\asyncio\events.py", line 660 in new_event_loop
  File "C:\projects\cpython\lib\asyncio\events.py", line 640 in get_event_loop
  File "<string>", line 2 in <module>
---
```

--- a/Doc/whatsnew/3.7.rst
+++ b/Doc/whatsnew/3.7.rst
@@ -733,6 +733,10 @@ include:
 * Exceptions occurring in cancelled tasks are no longer logged.
   (Contributed by Yury Selivanov in :issue:`30508`.)
 
+* New ``WindowsSelectorEventLoopPolicy`` and
+  ``WindowsProactorEventLoopPolicy`` classes.
+  (Contributed by Yury Selivanov in :issue:`33792`.)
+
 Several ``asyncio`` APIs have been
 :ref:`deprecated <whatsnew37-asyncio-deprecated>`.
 

--- a/Lib/asyncio/windows_events.py
+++ b/Lib/asyncio/windows_events.py
@@ -21,7 +21,8 @@ from .log import logger
 
 __all__ = (
     'SelectorEventLoop', 'ProactorEventLoop', 'IocpProactor',
-    'DefaultEventLoopPolicy',
+    'DefaultEventLoopPolicy', 'WindowsSelectorEventLoopPolicy',
+    'WindowsProactorEventLoopPolicy',
 )
 
 
@@ -801,8 +802,12 @@ class _WindowsSubprocessTransport(base_subprocess.BaseSubprocessTransport):
 SelectorEventLoop = _WindowsSelectorEventLoop
 
 
-class _WindowsDefaultEventLoopPolicy(events.BaseDefaultEventLoopPolicy):
+class WindowsSelectorEventLoopPolicy(events.BaseDefaultEventLoopPolicy):
     _loop_factory = SelectorEventLoop
 
 
-DefaultEventLoopPolicy = _WindowsDefaultEventLoopPolicy
+class WindowsProactorEventLoopPolicy(events.BaseDefaultEventLoopPolicy):
+    _loop_factory = ProactorEventLoop
+
+
+DefaultEventLoopPolicy = WindowsSelectorEventLoopPolicy

--- a/Lib/test/test_asyncio/test_windows_events.py
+++ b/Lib/test/test_asyncio/test_windows_events.py
@@ -166,5 +166,36 @@ class ProactorTests(test_utils.TestCase):
         fut.cancel()
 
 
+class WinPolicyTests(test_utils.TestCase):
+
+    def test_selector_win_policy(self):
+        async def main():
+            self.assertIsInstance(
+                asyncio.get_running_loop(),
+                asyncio.SelectorEventLoop)
+
+        old_policy = asyncio.get_event_loop_policy()
+        try:
+            asyncio.set_event_loop_policy(
+                asyncio.WindowsSelectorEventLoopPolicy())
+            asyncio.run(main())
+        finally:
+            asyncio.set_event_loop_policy(old_policy)
+
+    def test_proactor_win_policy(self):
+        async def main():
+            self.assertIsInstance(
+                asyncio.get_running_loop(),
+                asyncio.ProactorEventLoop)
+
+        old_policy = asyncio.get_event_loop_policy()
+        try:
+            asyncio.set_event_loop_policy(
+                asyncio.WindowsProactorEventLoopPolicy())
+            asyncio.run(main())
+        finally:
+            asyncio.set_event_loop_policy(old_policy)
+
+
 if __name__ == '__main__':
     unittest.main()

--- a/Misc/NEWS.d/next/Library/2018-06-07-12-38-12.bpo-33792.3aKG7u.rst
+++ b/Misc/NEWS.d/next/Library/2018-06-07-12-38-12.bpo-33792.3aKG7u.rst
@@ -1,0 +1,2 @@
+Add asyncio.WindowsSelectorEventLoopPolicy and
+asyncio.WindowsProactorEventLoopPolicy.

